packages/lib/services/joplinCloudUtils.ts

```diff
@@ -0,0 +1,104 @@
+import { Reducer } from 'react';
+import Setting from '../models/Setting';
+import { ApplicationPlatform, ApplicationType } from '../types';
+import shim from '../shim';
+import { _ } from '../locale';
+import { reg } from '../registry';
+
+type Events = 'LINK_USED' | 'COMPLETED';
```

Comment:
Types should usually be singular unless it represents a collection of objects or values. Also it's called "Events" here but "action" below so it needs to be one or the other. I guess "Action" is the right term for a reducer?

If there's any doubt it's better to stick to React's terms, which in this case uses state/action.